### PR TITLE
PHP8 compatibility

### LINKED ISSUE #36: xdebug.collect_params is removed in xdebug3
https://github.com/ingenerator/kohana-core/blob/a0bce0c7ba9e40f539d38cd03f03422c9af185b4/classes/Kohana/Core.php#L236

From https://xdebug.org/docs/upgrade_guide#changed-xdebug.collect_params

```xdebug.collect_params #
Has been removed. Arguments are now always visible with variable contents and argument name in Development Aids and Function Trace output.
```

xdebug3 is now latest stable and is required for php8 support

### PR DESCRIPTION
**Pull request by craig410**

### CODE REVIEW

**Comment by craig410** [comment]
Fixes #36

**Comment by craig410** [comment]
@acoulton think this is good to go if you agree. Can't run it on phpunit 9 for now, potentially because of the way that koharness sets it up but I'm not familiar enough with it to fix it. It's maybe not a problem, it should be compatible with phpunit 9 when we want to solve that although phpunit 8 reached end of support February 5, 2021.

File: tests/HorribleDeprecationWorkarounds.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+use PHPUnit\Framework\Assert;
+
+/**
+ * @author    Craig Gosman <craig@ingenerator.com>
+ * @licence   proprietary
+ *
+ *
+ * This is hideous so is self contained in a file and uses traits so we can find them and refactor in the future
+ */
+trait Workarounds
```
Comment (acoulton) [inline tests/HorribleDeprecationWorkarounds.php]: Could we call this something specific - `ObjectInternalAccessTestWorkarounds` or something? Just so we remember what it is and so we're not tempted to add other types of workaround to it in future?

**Review by acoulton — CHANGES_REQUESTED** [review]
Looks good, if you could rename the trait it'll be perfect. Obv needs a brief changelog mention too.

File: tests/HorribleDeprecationWorkarounds.php (hunk 1 above)
Comment (craig410) [reply tests/HorribleDeprecationWorkarounds.php]: I will. It wont stop us adding more traits under different names to workaround things in future though ;)
Haven't written the changelog yet was waiting to see if we're going for a point release on this. What do you reckon re phpunit? Is it worth a look at incase I'm just missing something obvious or will we ignore it for now?

Comment (acoulton) [reply tests/HorribleDeprecationWorkarounds.php]: Absolutely, that was what I was meaning :) Best to keep the different types of unpleasant workarounds separate.

I think don't worry about phpunit 9 just now, we can revisit later. There's no security implications to the EOL and if it passes on 8 now it should continue to.

Comment (craig410) [reply tests/HorribleDeprecationWorkarounds.php]: Grand I'll release it now and it might make the dep build at 9

**Pull request merged**